mithril-stm/src/multi_sig.rs

```diff
@@ -410,43 +384,15 @@ impl Signature {
             scalars.extend_from_slice(hasher.finalize().as_slice());
         }
 
-        let transmuted_vks: Vec<blst_p2> = vks
-            .iter()
-            .map(|vk| unsafe {
-                let mut projective_p2 = blst_p2::default();
-                blst_p2_from_affine(
-                    &mut projective_p2,
-                    &std::mem::transmute::<BlstVk, blst_p2_affine>(vk.0),
-                );
-                projective_p2
-            })
-            .collect();
-
-        let transmuted_sigs: Vec<blst_p1> = signatures
-            .iter()
-            .map(|&sig| unsafe {
-                let mut projective_p1 = blst_p1::default();
-                blst_p1_from_affine(
-                    &mut projective_p1,
-                    &std::mem::transmute::<BlstSig, blst_p1_affine>(sig),
-                );
-                projective_p1
-            })
-            .collect();
+        let transmuted_vks: Vec<blst_p2> = vks.iter().map(vk_from_p2_affine).collect();
+
+        let transmuted_sigs: Vec<blst_p1> = signatures.iter().map(sig_to_p1).collect();
 
         let grouped_vks = p2_affines::from(transmuted_vks.as_slice());
         let grouped_sigs = p1_affines::from(transmuted_sigs.as_slice());
 
-        let aggr_vk: BlstVk = unsafe {
-            let mut affine_p2 = blst_p2_affine::default();
-            blst_p2_to_affine(&mut affine_p2, &grouped_vks.mult(scalars.as_slice(), 128));
-            std::mem::transmute::<blst_p2_affine, BlstVk>(affine_p2)
-        };
-        let aggr_sig: BlstSig = unsafe {
-            let mut affine_p1 = blst_p1_affine::default();
-            blst_p1_to_affine(&mut affine_p1, &grouped_sigs.mult(scalars.as_slice(), 128));
-            std::mem::transmute::<blst_p1_affine, BlstSig>(affine_p1)
-        };
+        let aggr_vk: BlstVk = p2_affine_to_vk(&grouped_vks, scalars.as_slice());
+        let aggr_sig: BlstSig = p1_affine_to_sig(&grouped_sigs, scalars.as_slice());
```

Comment:
Here you can pass & instead of as_slice: 
```suggestion
        let aggr_vk: BlstVk = p2_affine_to_vk(&grouped_vks, &scalars);
        let aggr_sig: BlstSig = p1_affine_to_sig(&grouped_sigs, &scalars);
```